manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -303,9 +323,17 @@ func setReadModulesEndpoints(applicationContext *api.FybrikApplication, flows []
 // reconcile receives either FybrikApplication CRD
 // or a status update from the generated resource
 func (r *FybrikApplicationReconciler) reconcile(applicationContext *api.FybrikApplication) (ctrl.Result, error) {
-	utils.PrintStructure(applicationContext.Spec, r.Log, "FybrikApplication")
-	// Data User created or updated the FybrikApplication
+	uuid := utils.GetFybrikApplicationUUID(applicationContext)
 
+	// Log the request received - i.e. the fybrikapplication.spec
+	jsonStruct, err := json.Marshal(applicationContext.Spec)
+	if err != nil {
+		r.Log.Warn().Str(logging.FYBRIKAPPUUID, uuid).Msg("Failed converting fybrikapplication.spec to json")
+	} else {
+		r.Log.Trace().Bool(logging.FORUSER, true).Bool(logging.AUDIT, true).Str(logging.FYBRIKAPPUUID, uuid).RawJSON("FybrikApplication.Spec", jsonStruct)
+	}
```

Comment:
Can this functionality be extracted into a function, making the resource general? Something in the lines of utils.PrintStructure that received interface{} and a string.

Reply:
Yes, done